include/envoy/config/extension_config_provider.h

```diff
@@ -38,10 +38,9 @@ template <class Factory, class FactoryCallback> class ExtensionConfigProvider {
    * Validate that the configuration is applicable in the context of the provider. If an exception
    * is thrown by any of the config providers for an update, the extension configuration update is
    * rejected.
-   * @param proto_config is the candidate configuration update.
-   * @param factory used to instantiate an extension config.
+   * @param type_url is the candidate configuration protobuf type URL.
    */
-  virtual void validateConfig(const ProtobufWkt::Any& proto_config, Factory& factory) PURE;
+  virtual void validateConfig(const std::string& type_url) PURE;
```

Comment:
This has moved towards something more concrete from a more general set of params. Do you anticipate this being sufficient for future needs and hence are now restricting it? Is the only thing we can do here is make some predicate on type URL?

Reply:
Since the code is generic for all HTTP filters, not sure what else we can do besides type URL. I was also trying to save on not having to store the protobuf in the subscription, which can get expensive.